partner_lastname_uppercase/models/res_partner.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2023 Coop IT Easy SC
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def _convert_lastnames_to_uppercase(self):
+        return (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(
+                "partner_lastname_uppercase.convert_lastnames_to_uppercase", False
+            )
+        )
+
+    @api.model
+    def uppercase_all_lastnames(self):
+        individuals = self.search(
```

Comment:
```suggestion
        individuals = self.with_context(active_test=False).search(
```

Allow to include archived partners.